pyClient/zeth/utils.py

```diff
@@ -232,8 +233,11 @@ def encode_to_hash(message_list: Any) -> bytes:
         m_hex = m
 
         # Convert it into a hex
-        if isinstance(m, int):
-            m_hex = "{0:0>4X}".format(m)
+        if isinstance(m, (FQ, int)):
+            assert int(m) < 2**256, "The input cannot be encoded as " + \
+                "a bytes32 (longer than 256 bits)"
+
+            m_hex = "{0:0>64X}".format(int(m))
         elif isinstance(m, str) and (m[1] == "x"):
```

Comment:
`elif isinstance(m, str) and (m[1] == "x"):` -> `elif isinstance(m, str) and (m[0:2] == "0x"):` if you want to base your test on the `0x` prefix of an hexadecimal string.

Reply:
fixed